drivers/power/ltc3350/ltc3350.c

```diff
@@ -0,0 +1,371 @@
+	case 4:
+
+		ret = ltc3350_read_device_data(dev, LTC3350_AD_MEAS_VCAP4, value);
+		break;
+
+	default:
+		ret = -EINVAL;
+		break;
+	}
+
+	return ret;
+}
+
+#endif
+
+#if LTC3350_USE_ALARMS
+
+/***************************************************************************//**
+ * @brief Set the selected alarm and its related threshold
+ *
+ * @param dev          		- The device structure.
+ * @param alarm 			- Alarm to be setted.
+ * @param alarm_threshold   - The desired threshold value to the selected alarm.
+ *							  mV , Deg Celsius or uF
+ * @return ret         		- Result of the alarm setting operation.
+*******************************************************************************/
+int ltc3350_set_alarm(struct ltc3350_dev *dev, enum ltc3350_enum_alarms alarm,
+		      uint16_t alarm_threshold)
+{
+	int ret = 0;
+	uint16_t alarm_mask;
+
+	//Read register
+	ret = ltc3350_read_device_data(dev, LTC3350_AD_MSK_ALARMS, &alarm_mask);
+
+	if (ret)
+		return ret;
+
+	alarm_mask |= NO_OS_BIT(alarm);
+
+	//Write register
+	ret = ltc3350_write_device_data(dev, LTC3350_AD_MSK_ALARMS, alarm_mask);
+
+	if (ret)
+		return ret;
+
+	ret = ltc3350_write_device_data(dev, LTC3350_AD_CAP_UV_LVL + alarm,
+					alarm_threshold);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Mute an specific alarm
+ *
+ * @param dev          - The device structure.
+ * @param alarm 	   - Alarm to be muted.
+ *
+ * @return ret         - Result of the writing procedure.
+*******************************************************************************/
+int ltc3350_mute_alarm(struct ltc3350_dev *dev, enum ltc3350_enum_alarms alarm)
+{
+	int ret = 0;
+	uint16_t alarm_mask;
+
+	ret = ltc3350_read_device_data(dev, LTC3350_AD_MSK_ALARMS, &alarm_mask);
+
+	if (ret)
+		return ret;
+
+	alarm_mask &= ~NO_OS_BIT(alarm);
+
+	ret = ltc3350_write_device_data(dev, LTC3350_AD_MSK_ALARMS, alarm_mask);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Get alarms status mask
+ *
+ * @param dev          - The device structure.
+ * @param status	   - Status mask of the alarms
+ *
+ * @return ret         - Result of the reading procedure.
+*******************************************************************************/
+int ltc3350_get_alarm_status_mask(struct ltc3350_dev *dev, uint16_t *alarm_mask)
+{
+	int ret = 0;
+	uint16_t read_val;
+
+	ret = ltc3350_read_device_data(dev, LTC3350_AD_ALARM_REG, &read_val);
+
+	if (ret)
+		return ret;
+
+	*alarm_mask = read_val;
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Clear alarms status mask
+ *
+ * @param dev          - The device structure.
+ *
+ * @return ret         - Result of the writing procedure.
+*******************************************************************************/
+int ltc3350_clear_alarm_status_mask(struct ltc3350_dev *dev)
+{
+	int ret = 0;
```

Comment:
remove `ret` variable completely and just return the value of `ltc3350_write_device_data(dev, LTC3350_AD_CLR_ALARMS, 0x0000)`